modules/renter/upload.go

```diff
@@ -102,6 +103,10 @@ func (r *Renter) Upload(up modules.FileUploadParams) error {
 	lockID := r.mu.Lock()
 	defer r.mu.Unlock(lockID)
 
+	if filepath.Ext(up.Filename) != filepath.Ext(up.Nickname) {
+		return errors.New("nickname and file name must have the same extension")
+	}
```

Comment:
I dunno. Definitely feels like a front-end thing to me. Why do you want to restrict it?